grapple/types/structures.py

```diff
@@ -109,3 +110,151 @@ def serialize(value):
+        of_type, graphene_django.DjangoObjectType
+    ):
+        raise TypeError(
+            f"{of_type} is not a subclass of DjangoObjectType and it "
+            "cannot be used with QuerySetList."
+        )
+
+    # Enable page for Django Paginator.
+    if "page" not in kwargs:
+        kwargs["page"] = graphene.Argument(
+            PositiveInt,
+            default_value=1,
+            description=_("Page of resulting objects to return."),
+        )
+
+    # Enable per_page for Django Paginator.
+    if "per_page" not in kwargs:
+        kwargs["per_page"] = graphene.Argument(
+            PositiveInt,
+            default_value=10,
+            description=_("The maximum number of items to include on a page."),
+        )
+
+    # Enable ordering of the queryset
+    if enable_order is True and "order" not in kwargs:
+        kwargs["order"] = graphene.Argument(
+            graphene.String, description=_("Use Django ordering format.")
+        )
+
+    # If type is provided as a lazy value (e.g. using lambda), then
+    # the search has to be enabled explicitly.
+    if (enable_search is True and not inspect.isclass(of_type)) or (
+        enable_search is True
+        and inspect.isclass(of_type)
+        and class_is_indexed(of_type._meta.model)
+        and "search_query" not in kwargs
+    ):
+        kwargs["search_query"] = graphene.Argument(
+            graphene.String, description=_("Filter the results using Wagtail's search.")
+        )
+
+    if "id" not in kwargs:
+        kwargs["id"] = graphene.Argument(graphene.ID, description=_("Filter by ID"))
+
+    class PaginatedType(BasePaginatedType):
+        items = graphene.List(of_type, required=required)
+
+        class Meta:
+            name = type_name + "PaginatedType"
+
+    return graphene.Field(PaginatedType, **kwargs)
+
+
+def CursorQuerySet(of_type, type_class, **kwargs):
+    """
+    Paginated QuerySet type with arguments used by Django's query sets.
+
+    This type setts the following arguments on itself:
+
+    * ``first``
+    * ``last``
+    * ``before``
+    * ``after``
+
+    :param enable_search: Enable search query argument.
+    :type enable_search: bool
+    :param enable_order: Enable ordering via query argument.
+    :type enable_order: bool
+    """
+
+    type_name = type_class if isinstance(type_class, str) else type_class.__name__
+    type_name = type_name.lstrip("Stub")
+
+    # Check if the type is a Django model type. Do not perform the
+    # check if value is lazy.
+    if inspect.isclass(of_type) and not issubclass(
+        of_type, graphene_django.DjangoObjectType
+    ):
+        raise TypeError(
+            f"{of_type} is not a subclass of DjangoObjectType and it "
+            "cannot be used with QuerySetList."
+        )
+
+    class ModelCursor(graphene_django.DjangoObjectType):
+        class Meta:
+            name = type_name + "CursorType"
+            model = type_class
```

Comment:
@zerolab @NathHorrigan 
I'm getting this error `django.core.exceptions.AppRegistryNotReady: Models aren't loaded yet.` when using `register_cursor_query_field` decorator.

I would appreciate it if you can give me some advice on how to lazy load a Django Model or something similar to fix this issue 🙂 .

Reply:
Sorry for the delay @ruisaraiva19. Busy week here. Will try to look into it over the weekend. I would say separate this PR in two -- one for the Django pagination bit, and one for the cursor pagination

Reply:
Sounds good 👍.

Reply:
Updated this PR and opened #103 for the cursor pagination support